src/auto_scheduler/transform_step.cc

```diff
@@ -111,6 +115,10 @@ void StepApplyToState(const Step& step, State* state, const ComputeDAG& dag) {
     ps->ApplyToState(state);
   } else if (auto ps = step.as<SplitStepNode>()) {
     ps->ApplyToState(state);
+  } else if (auto ps = step.as<FollowSplitStepNode>()) {
```

Comment:
How  about to rewrite this as:

if (step.as<AnnotationStepNode>() or step.as<FuseStepNode>() or ... ) {
    ps->ApplyToState(state);
}

Maybe a little bit more compact?

Reply:
The `ps`s in different branches have different types, so they cannot be merged.

Reply:
Yes, the problem is these functions may have different parameters/return values, so they're not able to be merged to a single virtural function interface.